### DISCUSSION TITLE
Add redis prefix environment variable to docker image

### DISCUSSION THREAD

**Discussion by Shalashtein** [chatwoot/chatwoot#8449] [Self hosted]
### Is your feature or enhancement related to a problem? Please describe.

Managing chatwoot alongside our teck stack isn't ideal, since chatwoot requires its own redis instance, I'd like chatwoot to use our managed redis instances, but it would need to have a prefix environment variable so that it doesn't collide with our other apps

### Describe the solution you'd like

Add environment variable REDIS_PREFIX

### Describe alternatives you've considered

Hosting an instance specifically for chatwoot, and allowing chatwoot to use one of our managed redis instances at root level since the other apps all had prefixes

**Comment by pranavrajs** [comment]
@Shalashtein Can you please provide examples of the self-hosted apps with the support? I'm not sure if I fully understand the ask here.

**Reply by carlos-ramos-br** [reply]
N8N supports the inclusion of prefix.

**Comment by youssefsiam38** [comment]
Hi @pranavrajs,

The REDIS_PREFIX environment variable would help us use a single Redis server for multiple clients by allowing each application to have unique key prefixes, preventing key collisions.

For example:
App A uses the prefix app_a_
App B uses the prefix app_b_

With REDIS_PREFIX, Chatwoot could use a prefix like chatwoot_, ensuring its keys don't interfere with other applications’ keys in our shared Redis instance. This would avoid conflicts and streamline our infrastructure management.
